src/main/java/frc/robot/Robot.java

```diff
@@ -4,68 +4,258 @@
+        setNamespaceTestingCommands();
+        autoChooser = new AutoChooser(
+                new RootNamespace("auto chooser"),
+                new PlanBWindow(drivetrain).getCommand(), "plan b window",
+                new PlanBEdge(drivetrain).getCommand(), "plan b edge",
+                new SplooshAndVamooseWindow(drivetrain).getCommand(), "sploosh and vamoose",
+                new SmashAndDash(drivetrain).getCommand(), "smash and dash"
+        );
+        firstJoint.configureEncoders();
+        secondJoint.configureEncoders();
+        vision.setBackLimelightPipeline(VisionService.LimelightPipeline.HIGH_RRT);
+        vision.setFrontLimelightPipeline(VisionService.LimelightPipeline.HIGH_RRT);
+        userCommand = new InstantCommand(() -> {
+            drivetrain.setMode(CANSparkMax.IdleMode.kCoast);
+            firstJoint.setIdleMode(CANSparkMax.IdleMode.kCoast);
+            secondJoint.setIdleMode(CANSparkMax.IdleMode.kCoast);
+        }).ignoringDisable(true);
+    }
+
+    @Override
+    public void robotPeriodic() {
+        CommandScheduler.getInstance().run();
+        namespace.update();
+        drivetrain.periodic();
+        firstJoint.periodic();
+        secondJoint.periodic();
+        gripper.periodic();
+        vision.periodic();
+        leds.periodic();
+        SmashAndDash.update();
+        if (RobotController.getUserButton()) {
+            userCommand.schedule();
+        }
+    }
 
-  }
+    @Override
+    public void disabledInit() {
+        CommandScheduler.getInstance().cancelAll();
+        firstJoint.finish();
+        secondJoint.finish();
+        drivetrain.finish();
+        new InstantCommand(() -> {
+            firstJoint.setIdleMode(CANSparkMax.IdleMode.kBrake);
+            secondJoint.setIdleMode(CANSparkMax.IdleMode.kBrake);
+        }).ignoringDisable(true).schedule();
+        new OpenGripper(gripper).ignoringDisable(true).schedule();
+    }
 
-  @Override
-  public void robotPeriodic() {
-    CommandScheduler.getInstance().run();
-  }
+    @Override
+    public void disabledPeriodic() {
 
-  @Override
-  public void disabledInit() {
+    }
 
-  }
+    @Override
+    public void autonomousInit() {
+        new InstantCommand(() -> {
+            firstJoint.setIdleMode(CANSparkMax.IdleMode.kBrake);
+            secondJoint.setIdleMode(CANSparkMax.IdleMode.kBrake);
+        });
+        new CloseGripper(gripper).schedule();
+        CommandBase auto = null;
+        auto = new SmashAndDash(drivetrain).getCommand();
+//        auto = new ClimbPlanB(drivetrain);
+//        auto = new PlanBWindow(drivetrain).getCommand();
+//        auto = new PlanBEdge(drivetrain).getCommand();
+        if (auto != null && firstJoint.encoderConnected() && secondJoint.encoderConnected()) auto.schedule();
+        else new DriveArcade(drivetrain, 0.5, 0).withTimeout(2).schedule();
+    }
 
-  @Override
-  public void disabledPeriodic() {
+    @Override
+    public void autonomousPeriodic() {
 
```

Comment:
![y_tho_meme](https://github.com/Spikes-2212-Programming-Guild/ChargedUp2023/assets/71877371/694418c6-7536-438a-98ca-535127ef96e2)

Reply:
this has to be implememented